spec/mempool/gossip/dog.md

```diff
@@ -0,0 +1,606 @@
+> `HaveTx` messages, leading nodes to repeatedly sending `HaveTx` and `Reset` messages. On testnets,
+> we have observed that nodes continue to operate normally, except that bandwidth does not decrease
+> as expected.
+
+## Parameters
+
+The following parameters must be configured by each node at initialization.
+
+* `TargetRedundancy`: the desired redundancy level that the controller aims to keep as target
+  (within specified bounds).
+    ```bluespec "params" +=
+    const TargetRedundancy: int
+    ```
+    A target equal to 0 partially disables the Redundancy Control mechanism: the controller can
+    block `HaveTx` messages but cannot send `Reset` messages. Zero redundancy minimizes bandwidth
+    usage, achieving the lowest possible message overhead. In non-Byzantines networks, this is the
+    best possible scenario. However, in Byzantine networks it could potentially render nodes
+    isolated from transaction data. Therefore, the target should be set to a value greater than 0.
+    Experimental results suggest a value between 0.5 and 1, which is a safe number that does not
+    result in excessive duplicate transactions.
+
+    > Note: `TargetRedundancy` should ideally be specified as a real number, but reals are not
+    > currently supported by Quint.
+
+* `TargetRedundancyDeltaPercent`: a percentage (a number in the range `[0, 100)`) of
+  `TargetRedundancy` that defines acceptable lower and upper bounds for redundancy levels as a
+  deviation from the target value.
+    ```bluespec "params" +=
+    const TargetRedundancyDeltaPercent: int
+    ```
+    From this value the protocol derives the constants:
+    ```bluespec "params" +=
+    val _delta = TargetRedundancy * TargetRedundancyDeltaPercent / 100
+    val lowerBound = TargetRedundancy - _delta
+    val upperBound = TargetRedundancy + _delta
+    ```
+    This range provides flexibility, allowing the redundancy level to fluctuate while staying within
+	tolerable limits. For example, a `TargetRedundancy` of 0.5 with a `TargetRedundancyDeltaPercent`
+	of 20% would allow a redundancy level between 0.4 and 0.6.
+    
+    Based on experimentation, a `TargetRedundancyDeltaPercent` of around 20% strikes a good balance
+	between adaptability and stability.
+
+    > Note: Similar to `TargetRedundancy`, this parameter could also be defined as a real type if it
+    > were allowed by Quint.
+
+* `adjustInterval`: the time (in milliseconds) that the controller waits between successive
+  calls of `adjustRedundancy`.
+    ```bluespec "params" +=
+    const adjustInterval: int
+    ```
+	This interval should allow sufficient time for control messages (`HaveTx` and `Reset`) to
+	propagate through the network and take effect.
+	
+    A minimum value for `adjustInterval` depends on the network’s round-trip time (RTT). Assuming that
+    global latency typically stays below 500ms (maybe an excessive number, need references here), the
```

Comment:
500ms is not exaggerated. We should not think about network-level latencies, which are rarely higher than that, but on end-to-end latency. This includes all the processing, encoding/decoding, multiplexing, and protocol overhead. Plus the fact that a node does not only process transactions, it receives a bunch of other messages.

Being conservative here is helpful.

Reply:
Do you think then 1000ms is not enough for an end-to-end RTT? That is, should we increase the minimal value `adjustInterval` to 1500ms, 2000ms, or more?

Reply:
Below it writes 1000ms. I was thinking of using this as default value. Or maybe 1500ms to be sure.